package.json

```diff
@@ -78,66 +78,70 @@
-		"@wordpress/notices": "4.3.5",
-		"@wordpress/nux": "7.1.4",
-		"@wordpress/plugins": "6.3.11",
-		"@wordpress/preferences": "3.12.11",
-		"@wordpress/preferences-persistence": "1.27.1",
-		"@wordpress/primitives": "3.33.1",
-		"@wordpress/priority-queue": "2.35.1",
-		"@wordpress/private-apis": "0.17.2",
-		"@wordpress/redux-routine": "4.35.1",
-		"@wordpress/reusable-blocks": "4.12.15",
-		"@wordpress/rich-text": "6.12.8",
-		"@wordpress/router": "0.4.2",
-		"@wordpress/server-side-render": "4.12.13",
-		"@wordpress/shortcode": "3.35.1",
-		"@wordpress/style-engine": "1.18.1",
-		"@wordpress/token-list": "2.35.1",
-		"@wordpress/url": "3.36.1",
-		"@wordpress/viewport": "5.12.5",
-		"@wordpress/warning": "2.35.1",
-		"@wordpress/widgets": "3.12.15",
-		"@wordpress/wordcount": "3.35.1",
+		"@wordpress/a11y": "3.42.0",
+		"@wordpress/annotations": "2.42.0",
+		"@wordpress/api-fetch": "6.39.0",
+		"@wordpress/autop": "3.42.0",
+		"@wordpress/blob": "3.42.0",
+		"@wordpress/block-directory": "4.19.0",
+		"@wordpress/block-editor": "12.10.0",
+		"@wordpress/block-library": "8.19.0",
+		"@wordpress/block-serialization-default-parser": "4.42.0",
+		"@wordpress/blocks": "12.19.0",
+		"@wordpress/commands": "0.13.0",
+		"@wordpress/components": "25.8.0",
+		"@wordpress/compose": "6.19.0",
+		"@wordpress/core-commands": "0.11.0",
+		"@wordpress/core-data": "6.19.0",
+		"@wordpress/customize-widgets": "4.19.0",
+		"@wordpress/data": "9.12.0",
+		"@wordpress/data-controls": "3.11.0",
+		"@wordpress/date": "4.42.0",
+		"@wordpress/deprecated": "3.42.0",
+		"@wordpress/dom": "3.42.0",
+		"@wordpress/dom-ready": "3.42.0",
+		"@wordpress/edit-post": "7.19.0",
+		"@wordpress/edit-site": "5.19.0",
+		"@wordpress/edit-widgets": "5.19.0",
+		"@wordpress/editor": "13.19.0",
+		"@wordpress/element": "5.19.0",
+		"@wordpress/escape-html": "2.42.0",
+		"@wordpress/format-library": "4.19.0",
+		"@wordpress/hooks": "3.42.0",
+		"@wordpress/html-entities": "3.42.0",
+		"@wordpress/i18n": "4.42.0",
+		"@wordpress/icons": "9.33.0",
+		"@wordpress/interactivity": "2.3.0",
+		"@wordpress/interface": "5.19.0",
+		"@wordpress/is-shallow-equal": "4.42.0",
+		"@wordpress/keyboard-shortcuts": "4.19.0",
+		"@wordpress/keycodes": "3.42.0",
+		"@wordpress/list-reusable-blocks": "4.19.0",
+		"@wordpress/media-utils": "4.33.0",
+		"@wordpress/notices": "4.10.0",
+		"@wordpress/nux": "8.4.0",
+		"@wordpress/patterns": "1.3.0",
+		"@wordpress/plugins": "6.10.0",
+		"@wordpress/preferences": "3.19.0",
+		"@wordpress/preferences-persistence": "1.34.0",
+		"@wordpress/primitives": "3.40.0",
+		"@wordpress/priority-queue": "2.42.0",
+		"@wordpress/private-apis": "0.24.0",
+		"@wordpress/redux-routine": "4.42.0",
+		"@wordpress/reusable-blocks": "4.19.0",
+		"@wordpress/rich-text": "6.19.0",
+		"@wordpress/router": "0.11.0",
+		"@wordpress/server-side-render": "4.19.0",
+		"@wordpress/shortcode": "3.42.0",
+		"@wordpress/style-engine": "1.25.0",
+		"@wordpress/sync": "0.4.0",
```

Comment:
I think the sync package should be removed from here for the moment. It's still experimental and all of its usage is behind a feature flag. I expect the web pack bundler to treeshake all of its usage.

Reply:
Looks like the problem might be in [getOrLoadEntitiesConfig](https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/entities.js#L398) which uses `registerSyncConfigs` without any flags. I think it we fix that we should be able to remove the sync package from core. But it would still be good to make the sync package private, or a dev dependency of core-data so that it doesn't get auto-included every time we run the package sync task in core.

Reply:
We could make the package to a bundled package for now like icons or interface maybe

Reply:
Whatever works best! As long as we can update the packages without breaking core 😄

Reply:
Hmm looking into this, making it a bundled package won't keep it out of core - interface, icons and the recent undo-manager are all in here (is adding undo-manager an expected change btw?). Also not sure how viable it is to make the package private given it's [already on npm](https://www.npmjs.com/package/@wordpress/sync). 

Perhaps if we add the experimental flags around the uses of `registerSyncConfigs` in [getOrLoadEntitiesConfig](https://github.com/WordPress/gutenberg/blob/trunk/packages/core-data/src/entities.js#L398) we can get the treeshaking to work properly? I'm not 100% sure that'll work to remove the dependency though 🤔

Reply:
[This PR](https://github.com/WordPress/gutenberg/pull/54710) is an attempt to fix the issue: it should at least be able to stop the editor from crashing if we remove the sync package from core. I'm not sure it will stop the package from being added as a dependency when `sync-gutenberg-packages` runs though. @gziolo do you have any ideas about how we could do this?

Reply:
https://github.com/WordPress/gutenberg/blob/e33798db47b70bd8de7a24b56cde58830ebcdb4a/packages/core-data/package.json#L46

I see that this package is listed as a dependency so it's going to be installed. It isn't listed as a bundled dependency so webpack will put it as a dependency:

https://github.com/WordPress/gutenberg/blob/e33798db47b70bd8de7a24b56cde58830ebcdb4a/packages/dependency-extraction-webpack-plugin/lib/util.js#L2-L6

> We could make the package to a bundled package for now like icons or interface maybe

That might work. We also could put the code behind `if ( process.env.IS_GUTENBERG_PLUGIN )` to ensure that it isn't bundled on production after all.

Reply:
Installing the package as a dependency of another package is not a problem if it's not exposed as an API. If tree shaking excludes the package from the bundle of "core-data", there's no need to do anything. Otherwise, we can consider making it a bundled package but we shouldn't be touching the package itself.

Reply:
At the moment, `wp-sync` is listed as a dependency for the production version of `wp-core-data`:

<img width="1412" alt="Screenshot 2023-09-22 at 20 50 05" src="https://github.com/WordPress/wordpress-develop/assets/699132/7f313c8c-bd39-46bf-92bf-df05950be3a3">

Let's see what happens when the changes get applied in the `@wordpress/core-data` to help with tree shaking. In fact, we can do some testing with the webpack config after Beta 1. For now, we just need to ensure that all editors load correctly 😄